_drafts/2024-05-18-get-started-with-aws-tf-01.md

```diff
@@ -0,0 +1,254 @@
+
+Create a new file called `aws-provider-key.tf` and add the following content:
+
+```terraform
+# /infra/aws-provider-key.tf
+
+provider "aws" {
+    access_key = ""
+    secret_key = ""
+    region = "ap-southeast-2"
+}
+```
+
+You can opt to use a region closer to you. As I'm in Australia I'm opting to use the australian region.
+
+Next you need to create two things in the AWS Management Console:
+1. Create a [new User](https://docs.aws.amazon.com/IAM/latest/UserGuide/id_users_create.html) with administrator permission
+2. Create an [access key](https://docs.aws.amazon.com/IAM/latest/UserGuide/id_credentials_access-keys.html) for the user
+
+Take the access key and corresponding secret key and add them to your provider key your defined earlier.
+
+> These API keys should be treated like credentials. Do not commit this file to source control. Otherwise the key could be discovered and you'll need to replace them.
+
+Now try running `terraform apply` once more and everything should be hunky dory.
+
+## Step 3 - Connect to our EC2 Instance
+
+If you've made it this far it means you've successfully configured terraform with an aws account and have deployed an EC2 instance to AWS. Nice!  
+Let's try connect to this instance.  
+We can find the ip address by logging in to the AWS Management Console and opening the instance details within EC2.
+Copy the IPv4 address and in a terminal run `ping <the-ip-address>`.
+
+No response...  
+
+This is because when we define the EC2 instance, a default security group is applied to it. This security group allows incoming connections on any port using any protocol but only from other aws resources with the same security group rules...
+
+If we want to connect to our instance, we need to make our own rules.
+
+When defining a security group rule, we declare if the rule is an ingress or egress rule (incoming or outgoing). We can also declare which port the rule applies to and the ip address range.
+
+We don't want just anyone to connect to this service so for now we'll only allow our local device's public ip address in the address range. You can find this easily using a site like [whatismyipaddress.com](www.whatismyipaddress.com). 
+
+Open `unicon-api.tf` and add the following resources, providing your public ip for `cidr_ipv4` value:
+
+```terraform
+# /infra/unicorn-api.tf
+
+resource "aws_security_group" "unicorn_api_security_group" {
+  name = "unicorn-api-security-group"
+}
+
+resource "aws_vpc_security_group_ingress_rule" "allow_from_local_device" {
+  security_group_id = aws_security_group.unicorn_api_security_group.id
+  description       = "allow ingress on any port from any ip"
+  cidr_ipv4         = "<YOUR_IP_ADDRESS_HERE>/32"
+  ip_protocol       = "tcp"
+  from_port         = 22
+  to_port           = 22
+}
+```
```

Comment:
Do we want to maybe throw these resources into their own file? or are you planning on providing instructions for a refactor later when the codebase gets bigger?

Totally up to you either way, just encouraging you to make this a conscious choice.